collect_app/src/main/java/org/odk/collect/android/preferences/screens/AdminPreferencesFragment.java

```diff
@@ -235,14 +231,22 @@ public CharSequence provideSummary(Preference preference) {
     }
 
     public void deleteProject() {
-        ProjectDeleter projectDeleter = new ProjectDeleter(projectsRepository, currentProjectProvider, formUpdateManager, instanceSubmitScheduler);
-        Project.Saved newProject = projectDeleter.deleteCurrentProject();
-
-        if (newProject != null) {
-            ActivityUtils.startActivityAndCloseAllOthers(requireActivity(), MainMenuActivity.class);
-            ToastUtils.showLongToast(getString(R.string.switched_project, newProject.getName()));
-        } else {
-            ActivityUtils.startActivityAndCloseAllOthers(requireActivity(), SplashScreenActivity.class);
+        DeleteProjectResult deleteProjectResult = projectDeleter.deleteCurrentProject();
+
+        if (deleteProjectResult instanceof DeleteProjectResult.UnsentInstances) {
```

Comment:
Maybe worth a Kotlin convert here to allow us to use `when`?

Reply:
I like converting into kotlin so I thought about it, the problem is that it extends another class (Java class) so I thought you wouldnt like it but if you are ok with it I can do that

Reply:
Should be fine? I think if it causes problems then yeah we can make it a separate PR, but I'm happy for a subclass of a Java class to use Kotlin

Reply:
Done.